src/main/java/org/itsallcode/openfasttrace/importer/asciidoc/AsciiDocImporter.java

```diff
@@ -1,26 +1,267 @@
+    private void processSpecificationItemComment(final StructuralNode block)
+    {
+        block.findBy(Map.of(KEY_ROLE, ROLE_COMMENT)).stream().findFirst()
+                .flatMap(this::getStringContent)
+                .ifPresent(listener::appendComment);
+    }
+
+    private void processSpecificationItemBlock(final String sid, final StructuralNode block)
+    {
+        final var specItemId = new SpecificationItemId.Builder(sid).build();
+        final var location = getLocation(block);
+        LOG.fine(() -> String.format("adding specification item [ID: %s, location: %s]", specItemId,
+                location));
+
+        this.listener.beginSpecificationItem();
+        this.listener.setId(specItemId);
+        this.listener.setLocation(location);
+        processSpecificationItemTitle(block);
+        processSpecificationItemDepends(block);
+        processSpecificationItemNeeds(block);
+        processSpecificationItemCovers(block);
+        processSpecificationItemDescription(block);
+        processSpecificationItemRationale(block);
+        processSpecificationItemComment(block);
+        this.listener.endSpecificationItem();
+    }
+
+    private void processForwardingBlock(final String skippedType, final StructuralNode block)
+    {
+        final var coveredSpecItems = getCoveredIds(block);
+        if (coveredSpecItems.isEmpty() || coveredSpecItems.size() > 1)
+        {
+            LOG.severe(
+                    () -> """
+                            Cannot process .specitem [%s]. A .specitem used to forward needs must have an \
+                            'oft-covers' attribute with exactly one spec item ID.
+                            """.formatted(getLocation(block)));
+            return;
+        }
+        final var coveredSid = new SpecificationItemId.Builder(coveredSpecItems.get(0)).build();
+        final var specItemId = new SpecificationItemId.Builder()
+                .artifactType(skippedType)
+                .name(coveredSid.getName())
+                .revision(coveredSid.getRevision())
+                .build();
+        final var location = getLocation(block);
+        LOG.fine(() -> "adding forwarding specification item [ID: %s, location: %s]".formatted(specItemId,
+                location));
+
+        this.listener.beginSpecificationItem();
+        this.listener.setId(specItemId);
+        this.listener.setLocation(location);
+        this.listener.addCoveredId(coveredSid);
+        processSpecificationItemNeeds(block);
+        this.listener.endSpecificationItem();
+    }
+
+    // [impl->dsn~adoc-specification-item-id~1]
+    // [impl->dsn~adoc-artifact-forwarding-notation~1]
+    private void processSpecificationItem(final StructuralNode block)
+    {
+        Optional.ofNullable(block.getAttribute(ATTRIBUTE_OFT_SID))
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .filter(s -> !s.isBlank())
```

Comment:
```suggestion
                .filter(attributeValue -> !attributeValue.isBlank())
```
Big fan of being explicit. :-)